### PR TITLE
Add testPerformingCancelledRequest

### PR DESCRIPTION
**Pull request by 8W9aG**
* Test cases where the request has been cancelled
during authorization

### CODE REVIEW

**Comment by 8W9aG** [comment]
@jgavris @rastersize @dflems @JensAyton

File: SPTDataLoaderTests/SPTDataLoaderRequestResponseHandlerMock.m [hunk 1]
```diff
@@ -44,4 +44,13 @@ - (void)receivedInitialResponse:(SPTDataLoaderResponse *)response
     self.lastReceivedResponse = response;
 }
 
+- (BOOL)shouldAuthoriseRequest:(SPTDataLoaderRequest *)request
+{
+    return self.authorising;
```
Comment (jgavris) [inline SPTDataLoaderTests/SPTDataLoaderRequestResponseHandlerMock.m]: use the getter!

**Comment by rastersize** [comment]
The test fails for OS X and tvOS (same issue).

```
SPTDataLoaderServiceTest
  testPerformingCancelledRequest, ((self.service.handlers.count) equal to (0u)) failed: ("1") is not equal to ("0") - There should be no handlers for an already cancelled request
  /Users/travis/build/spotify/SPTDataLoader/SPTDataLoaderTests/SPTDataLoaderServiceTest.m:356
```

**Comment by dflems** [comment]
lgtm :+1:

**Comment by rastersize** [comment]
:+1:

**Pull request merged**